test/unit/specs/core/executeCommandFactory.spec.js

```diff
@@ -116,29 +118,90 @@ describe("executeCommandFactory", () => {
+      logger,
+      configureCommand,
+      handleError,
+      validateCommandOptions: options => options
+    });
+    executeCommand("configure");
+    return executeCommand("test", {}).catch(() => {
+      expect(handleError).toHaveBeenCalledWith(
+        new Error("Unexpected error"),
+        "test"
+      );
+    });
+  });
+
+  it("executes component commands", () => {
+    const validateCommandOptionsSpy = jasmine
+      .createSpy()
+      .and.returnValues(
+        "with-result-post-validation-options",
+        "without-result-post-validation-options"
+      );
+    const testCommandWithResult = jasmine.createSpyObj(
+      "testCommandWithResult",
+      {
+        run: { foo: "bar" }
+      }
+    );
+    const testCommandWithoutResult = jasmine.createSpyObj(
+      "testCommandWithoutResult",
+      {
+        run: undefined
+      }
+    );
+    const componentRegistry = {
+      getCommand: jasmine
+        .createSpy("getCommand")
+        .and.returnValues(testCommandWithResult, testCommandWithoutResult),
+      getCommandNames() {
+        return ["testCommandWithResult", "testCommandWithoutResult"];
+      }
+    };
+    const configureCommand = () => Promise.resolve(componentRegistry);
     const executeCommand = executeCommandFactory({
       logger,
       configureCommand,
       handleError,
       validateCommandOptions: validateCommandOptionsSpy
     });
     executeCommand("configure");
-    return executeCommand("test", "pre-validation-options").then(() => {
+    return Promise.all([
+      executeCommand(
+        "testCommandWithResult",
+        "with-result-pre-validation-options"
+      ),
+      executeCommand(
+        "testCommandWithoutResult",
+        "without-result-pre-validation-options"
+      )
+    ]).then(results => {
+      expect(results[0]).toEqual({ foo: "bar" });
+      expect(results[1]).toEqual({});
+      expect(validateCommandOptionsSpy).toHaveBeenCalledWith({
+        command: testCommandWithResult,
+        options: "with-result-pre-validation-options"
+      });
       expect(validateCommandOptionsSpy).toHaveBeenCalledWith({
-        command: testCommand,
-        options: "pre-validation-options"
+        command: testCommandWithoutResult,
+        options: "without-result-pre-validation-options"
       });
-      expect(runCommandSpy).toHaveBeenCalledWith("post-validation-options");
+      expect(testCommandWithResult.run).toHaveBeenCalledWith(
+        "with-result-post-validation-options"
+      );
+      expect(testCommandWithoutResult.run).toHaveBeenCalledWith(
+        "without-result-post-validation-options"
+      );
     });
   });
 
   it("executes the core commands", () => {
     const configureCommand = jasmine
       .createSpy()
       .and.returnValue(Promise.resolve("configureResult"));
-    const setDebugCommand = jasmine
-      .createSpy()
-      .and.returnValue(Promise.resolve("logResult"));
+    const setDebugCommand = jasmine.createSpy();
```

Comment:
Not returning a promise or any result reflects the current reality of the `setDebugCommand`.